lib/macho/macho_file.rb

```diff
@@ -497,13 +504,13 @@ def set_lc_str_in_cmd(cmd, lc_str, old_str, new_str)
       @raw_data[cmd.offset + 4, 4] = [new_padded_size].pack(fmt)
 
       # delete the old str
-      @raw_data.slice!(cmd.offset + lc_str.to_i...cmd.offset + cmd.class.bytesize + old_str.size)
+      @raw_data.slice!(cmd.offset + lc_str.to_i...cmd.offset + cmd.class.bytesize + old_str.bytesize)
```

Comment:
While you're here, would you mind fixing the inconsistency that you're using `lc_str.to_i` on the left-hand side and `lc_str.class.bytesize` on the right-hand side? (I think both should be `lc_str.to_i` and just accidentally coincide in the cases we've come across so far.)

Better yet, use a different signature of `String#slice!` to avoid some repetition:

```ruby
@raw_data.slice!(cmd.offset + lc_str.to_i, old_str.bytesize)
```